src/services/zapper.js

```diff
@@ -0,0 +1,228 @@
+  switch (network) {
+    case 'ethereum': return CHAIN.ETHEREUM;
+    case 'polygon': return CHAIN.POLYGON;
+    case 'binance-smart-chain': return CHAIN.BINANCE;
+    case 'xdai': return CHAIN.XDAI;
+    default: return null;
+  }
+};
+
+
+const walletProtocols = [
+  PROTOCOLS.TOKENS,
+  PROTOCOLS.BITCOIN,
+  PROTOCOLS.UNIT, // Generic?
+  // PROTOCOLS.NFT, // we don't display NFT prices
+];
+
+const investmentProtocols = [
+  PROTOCOLS.AUTO_FARM,
+  PROTOCOLS.AAVE,
+  PROTOCOLS.AAVE_AMM,
+  PROTOCOLS.AAVE_V2,
+  PROTOCOLS.ALCHEMIX,
+  PROTOCOLS.ALPHA,
+  PROTOCOLS.BADGER,
+  PROTOCOLS.BARNBRIDGE,
+  PROTOCOLS.BEEFY,
+  PROTOCOLS.BZX,
+  PROTOCOLS.COMPOUND,
+  PROTOCOLS.CREAM,
+  PROTOCOLS.CURVE,
+  PROTOCOLS.DEFISAVER,
+  PROTOCOLS.DEVERSIFI,
+  PROTOCOLS.DHEDGE,
+  PROTOCOLS.DFORCE,
+  PROTOCOLS.ELLIPSIS,
+  PROTOCOLS.IDLE,
+  PROTOCOLS.HARVEST,
+  PROTOCOLS.HEGIC,
+  PROTOCOLS.KEEPER_DAO,
+  PROTOCOLS.MUSHROOM,
+  PROTOCOLS.PANCAKESWAP,
+  PROTOCOLS.PICKLE,
+  PROTOCOLS.RARI,
+  PROTOCOLS.REALT,
+  PROTOCOLS.RIBBON,
+  PROTOCOLS.SADDLE,
+  PROTOCOLS.SHELL,
+  PROTOCOLS.SMOOTHY,
+  PROTOCOLS.VALUE,
+  PROTOCOLS.VENUS,
+  PROTOCOLS.YAXIS,
+  PROTOCOLS.YEARN,
+];
+
+const liquidityPoolProtocols = [
+  PROTOCOLS.BALANCER,
+  PROTOCOLS.BANCOR,
+  PROTOCOLS.DODO,
+  PROTOCOLS.DYDX,
+  PROTOCOLS.FUTURESWAP,
+  PROTOCOLS.LINKSWAP,
+  PROTOCOLS.LOOPRING,
+  PROTOCOLS.LIQUITY,
+  PROTOCOLS.MOONISWAP,
+  PROTOCOLS.ONEINCH,
+  PROTOCOLS.QUICKSWAP,
+  PROTOCOLS.SFINANCE,
+  PROTOCOLS.SNOWSWAP,
+  PROTOCOLS.SPOOKYSWAP,
+  PROTOCOLS.SUSHISWAP,
+  PROTOCOLS.SWERVE,
+  PROTOCOLS.SYNTHETIX,
+  PROTOCOLS.UNISWAP,
+  PROTOCOLS.UNISWAP_V2,
+  PROTOCOLS.UNISWAP_V3,
+  PROTOCOLS.XSIGMA,
+];
+
+const depositProtocols = [
+  PROTOCOLS.BELLA,
+  PROTOCOLS.COVER, // Not sure about this one...
+  PROTOCOLS.DEFIVADEX, // Not sure... insurance mining?
+  PROTOCOLS.DSD, // Just a token
+  PROTOCOLS.ESD, // Another token
+  PROTOCOLS.MAKER, // DAI
+  PROTOCOLS.OTHER, // Optional?
+  PROTOCOLS.POOLTOGETHER,
+  PROTOCOLS.REFLEXER,
+  PROTOCOLS.TOKENSETS,
+  PROTOCOLS.VESPER,
+];
+
+const rewardProtocols = [];
+
+const supportedProtocols = [
+  ...walletProtocols,
+  ...investmentProtocols,
+  ...depositProtocols,
+  ...liquidityPoolProtocols,
+  ...rewardProtocols,
+];
+
+export const mapZapperProtocolIdToBalanceCategory = (protocol: string): ?string => {
+  if (walletProtocols.includes(protocol)) return 'wallet';
+  if (investmentProtocols.includes(protocol)) return 'investments';
+  if (liquidityPoolProtocols.includes(protocol)) return 'liquidityPools';
+  if (depositProtocols.includes(protocol)) return 'deposits';
+  if (rewardProtocols.includes(protocol)) return 'rewards';
+
+  return null;
+};
+
+export const getZapperProtocolBalanceOnNetwork = async (
+  addresses: string[],
+  protocol: string,
+  network: string,
+) => {
+  if (!supportedProtocols.includes(protocol)) return null;
+
+  try {
+    const result = await httpRequest.get(
```

Comment:
Do we need the try / catch here? Could we not use the .catch method of the promise `httpRequest.get`?